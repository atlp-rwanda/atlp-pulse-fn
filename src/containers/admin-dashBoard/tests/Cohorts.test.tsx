import { MockedProvider as ApolloProvider } from '@apollo/client/testing';
import '@testing-library/jest-dom';
import React from 'react';
import '../../../../test/jest/__mocks__/matchMedia';

import { act, fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AdminCohort from '../Cohorts';

describe('Cohorts tests', () => {
  it('Snapshots Cohorts Page', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <ApolloProvider>
            <AdminCohort />
          </ApolloProvider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  jest.setTimeout(10000);
  it('Renders Cohorts Page', async () => {
    render(
      <MemoryRouter>
        <ApolloProvider addTypename={false}>
          <AdminCohort />
        </ApolloProvider>
      </MemoryRouter>,
    );

    const updateIcon = await screen.findByTestId('updateIcon');
    const deleteIcon = await screen.findByTestId('deleteIcon');
    const viewTraineeIcon = await screen.findByTestId('traineeIcon');
    expect(updateIcon).toBeInTheDocument();
    expect(deleteIcon).toBeInTheDocument();
    expect(viewTraineeIcon).toBeInTheDocument();

    await act(async () => {
      fireEvent.click(updateIcon);
    });
    const updateCohortModal = await screen.findByTestId('updateCohortModal');
    const confirmUpdateBtn = await screen.findByTestId('confirmUpdateBtn');
    expect(updateCohortModal).toBeVisible();
    expect(confirmUpdateBtn).toBeInTheDocument();
    await act(async () => {
      fireEvent.click(confirmUpdateBtn);
    });

    await act(async () => {
      fireEvent.click(deleteIcon);
    });
    const confirmDeleteBtn = await screen.findByTestId('confirmDeleteBtn');
    expect(confirmDeleteBtn).toBeInTheDocument();
    await act(async () => {
      fireEvent.click(confirmDeleteBtn);
    });
  });
});
