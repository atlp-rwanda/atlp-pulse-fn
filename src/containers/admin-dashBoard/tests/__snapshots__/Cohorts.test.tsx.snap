// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohorts tests Snapshots Cohorts Page 1`] = `
Array [
  <div
    className="h-screen w-screen bg-black bg-opacity-30 backdrop-blur-sm absolute flex items-center justify-center px-4 hidden"
  >
    <div
      className="bg-white dark:bg-dark-bg w-screen md:w-1/2  xl:w-4/12 rounded-lg p-4 pb-8"
    >
      <div
        className="card-title w-full flex  flex-wrap justify-center items-center  "
      >
        <h3
          className="font-bold text-sm dark:text-white text-center w-11/12 uppercase"
        >
          Add Cohort
        </h3>
        <hr
          className=" bg-primary border-b my-3 w-full"
        />
      </div>
      <div
        className="card-body"
      >
        <form
          className=" py-3 px-8"
        >
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className="border border-primary rounded outline-none px-5 dark:bg-dark-frame-bg dark:text-white font-sans text-xs py-2 w-full"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="name"
                type="text"
              />
            </div>
          </div>
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className="border border-primary py-2 dark:bg-dark-frame-bg dark:text-white rounded outline-none px-5 font-sans text-xs w-full"
                name="phase"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Phase"
                type="text"
              />
            </div>
          </div>
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <div
                className="my-react-select-container css-b62m3t-container"
                onKeyDown={[Function]}
              >
                <span
                  className="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  className="css-1f43avz-a11yText-A11yText"
                />
                <div
                  className="my-react-select__control css-6zplfg-control"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <div
                    className="my-react-select__value-container css-319lph-ValueContainer"
                  >
                    <div
                      className="my-react-select__placeholder css-14el2xx-placeholder"
                      id="react-select-2-placeholder"
                    >
                      Select Coordinator
                    </div>
                    <div
                      className="my-react-select__input-container css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-2-placeholder"
                        aria-expanded={false}
                        aria-haspopup={true}
                        autoCapitalize="none"
                        autoComplete="off"
                        autoCorrect="off"
                        className="my-react-select__input"
                        disabled={false}
                        id="react-select-2-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        role="combobox"
                        spellCheck="false"
                        style={
                          Object {
                            "background": 0,
                            "border": 0,
                            "color": "inherit",
                            "font": "inherit",
                            "gridArea": "1 / 2",
                            "label": "input",
                            "margin": 0,
                            "minWidth": "2px",
                            "opacity": 1,
                            "outline": 0,
                            "padding": 0,
                            "width": "100%",
                          }
                        }
                        tabIndex={0}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="my-react-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      className="my-react-select__indicator-separator css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      className="my-react-select__indicator my-react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <svg
                        aria-hidden="true"
                        className="css-tj5bde-Svg"
                        focusable="false"
                        height={20}
                        viewBox="0 0 20 20"
                        width={20}
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="input my-5 h-9 "
          >
            <div
              className="my-react-select-container css-b62m3t-container"
              onKeyDown={[Function]}
            >
              <span
                className="css-1f43avz-a11yText-A11yText"
                id="react-select-3-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                className="css-1f43avz-a11yText-A11yText"
              />
              <div
                className="my-react-select__control css-6zplfg-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className="my-react-select__value-container css-319lph-ValueContainer"
                >
                  <div
                    className="my-react-select__placeholder css-14el2xx-placeholder"
                    id="react-select-3-placeholder"
                  >
                    Program Name
                  </div>
                  <div
                    className="my-react-select__input-container css-6j8wv5-Input"
                    data-value=""
                  >
                    <input
                      aria-autocomplete="list"
                      aria-describedby="react-select-3-placeholder"
                      aria-expanded={false}
                      aria-haspopup={true}
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      className="my-react-select__input"
                      disabled={false}
                      id="react-select-3-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      role="combobox"
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "color": "inherit",
                          "font": "inherit",
                          "gridArea": "1 / 2",
                          "label": "input",
                          "margin": 0,
                          "minWidth": "2px",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "100%",
                        }
                      }
                      tabIndex={0}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="my-react-select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    className="my-react-select__indicator-separator css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    className="my-react-select__indicator my-react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="css-tj5bde-Svg"
                      focusable="false"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className="border border-primary py-2 dark:bg-dark-frame-bg dark:text-white rounded outline-none px-5 font-sans text-xs w-full"
                name="startDate"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="StartingDate"
                type="date"
              />
            </div>
          </div>
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className=" border border-primary py-2 dark:bg-dark-frame-bg dark:text-white rounded outline-none px-5 font-sans text-xs w-full"
                name="endDate"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="ClosingDate"
                type="date"
              />
            </div>
          </div>
          <div
            className="w-full flex justify-between"
          >
            <button
              className="btn info sm w-[30%] md:w-1/4 text-sm font-sans"
              data-testid="remove"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
               
              Cancel
            </button>
            <button
              className="btn primary sm w-[30%] md:w-1/4 text-sm font-sans p-0"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
               
              Save
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="h-screen w-screen bg-black fixed bg-opacity-30 backdrop-blur-sm flex items-center justify-center z-50 overflow-auto mt-10 p-4 hidden"
    data-testid="updateCohortModal"
  >
    <div
      className="bg-white dark:bg-dark-bg w-screen md:w-1/2  xl:w-4/12 rounded-lg p-4 pb-8"
    >
      <div
        className="card-title w-full flex  flex-wrap justify-center items-center  "
      >
        <h3
          className="font-bold text-sm dark:text-white text-center w-11/12 uppercase"
        >
          Update Cohort
        </h3>
        <hr
          className=" bg-primary border-b my-3 w-full"
        />
      </div>
      <div
        className="card-body"
      >
        <form
          className=" py-3 px-8"
        >
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className="border border-primary rounded outline-none px-5 dark:bg-dark-frame-bg dark:text-white font-sans text-xs py-2 w-full"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="name"
                type="text"
              />
            </div>
          </div>
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className="border border-primary py-2 dark:bg-dark-frame-bg dark:text-white rounded outline-none px-5 font-sans text-xs w-full"
                name="phase"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Phase"
                type="text"
              />
            </div>
          </div>
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className="border border-primary py-2 dark:bg-dark-frame-bg dark:text-white rounded outline-none px-5 font-sans text-xs w-full"
                name="startDate"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="StartingDate"
                type="date"
              />
            </div>
          </div>
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className=" border border-primary py-2 dark:bg-dark-frame-bg dark:text-white rounded outline-none px-5 font-sans text-xs w-full"
                name="endDate"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="ClosingDate"
                type="date"
              />
            </div>
          </div>
          <div
            className="w-full flex justify-between"
          >
            <button
              className="btn info sm w-[30%] md:w-1/4 text-sm font-sans"
              data-testid="remove"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
               
              Cancel
            </button>
            <button
              className="btn primary sm w-[30%] md:w-1/4 text-sm font-sans p-0"
              data-testid="confirmUpdateBtn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
               
              Save
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="h-screen w-screen bg-black fixed bg-opacity-30 backdrop-blur-sm flex items-center justify-center z-50 overflow-auto mt-10 p-4 hidden"
  >
    <div
      className="bg-white dark:bg-dark-bg shadow-lg px-5 py-8 rounded-md w-[100%] mx-auto lg:w-[60%]"
    >
      <div
        className=" flex items-center justify-between pb-6"
      >
        <div>
          <h2
            className="text-gray-800 dark:text-white font-semibold text-xl"
          >
            Trainees list
          </h2>
          <input
            className="border-gray-300 dark:bg-dark-tertiary dark:text-white border py-2 mt-4 rounded outline-none px-5 font-sans text-xs w-52 md:w-96"
            defaultValue=""
            onChange={[Function]}
            placeholder="Filter"
          />
        </div>
        <div
          style={
            Object {
              "color": "red",
              "marginTop": "-1.2cm",
            }
          }
        >
          <button
            className="btn  sxm "
            data-testid="delete"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </div>
      </div>
      <div
        style={
          Object {
            "overflowX": "auto",
          }
        }
      >
        <table
          className="min-w-full leading-normal"
          role="table"
        >
          <thead>
            <tr
              role="row"
            >
              <th
                className=" thead"
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Name
              </th>
              <th
                className=" thead"
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Email
              </th>
              <th
                className=" thead"
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Rating
              </th>
              <th
                className=" thead"
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Cohort
              </th>
              <th
                className=" thead"
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                Program
              </th>
            </tr>
          </thead>
          <tbody
            role="rowgroup"
          >
            <tr
              className=" bg-white dark:bg-dark-bg} "
              role="row"
            >
              <td
                className="data-cell"
                role="cell"
              />
              <td
                className="data-cell"
                role="cell"
              />
              <td
                className="data-cell"
                role="cell"
              />
              <td
                className="data-cell"
                role="cell"
              />
              <td
                className="data-cell"
                role="cell"
              />
            </tr>
          </tbody>
        </table>
        <div>
          <table
            className="w-full mt-2"
          >
            <tfoot
              className="w-full py-2"
            >
              <tr
                className="w-full py-2"
              >
                <td
                  colSpan={5}
                >
                  <div
                    className="w-full justify-center flex mx-auto flex-row items-center overflow-x-auto"
                  >
                    <button
                      className="page mx-2 text-white rounded-circle appearance-none font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                      disabled={true}
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        className="w-5"
                        fill="none"
                        fontSize="sm"
                        stroke="currentColor"
                        strokeWidth={2}
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M11 15l-3-3m0 0l3-3m-3 3h8M3 12a9 9 0 1118 0 9 9 0 01-18 0z"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                        />
                      </svg>
                    </button>
                     
                    <button
                      className="page mx-2 text-white rounded-circle font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                      disabled={true}
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        className="w-5"
                        fill="none"
                        fontSize="sm"
                        stroke="currentColor"
                        strokeWidth={2}
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                        />
                      </svg>
                    </button>
                     
                    <div
                      className="flex flex-row justify-center w-full sm:text-[12px] items-center "
                    >
                      <span
                        className="inline-block mx-2"
                      >
                        Page
                         
                        <strong>
                          1
                           of
                           1
                        </strong>
                         
                      </span>
                      <span
                        className="inline-block mx-2"
                      >
                        | Go to page:
                         
                        <input
                          className=" outline-none border rounded-md appearance-none border-primary dark:bg-primary pl-1"
                          defaultValue={1}
                          onChange={[Function]}
                          style={
                            Object {
                              "border": "solid 0.1rem #4aa5be",
                              "height": "30px",
                              "paddingLeft": "1.2rem",
                              "width": "50px",
                            }
                          }
                          type="number"
                        />
                      </span>
                       
                      <select
                        className="px-1/2 font-raleway rounded-md border border-primary dark:bg-primary"
                        onChange={[Function]}
                        style={
                          Object {
                            "border": "solid 0.1rem #4aa5be",
                            "height": "30px",
                          }
                        }
                        value={3}
                      >
                        <option
                          value={3}
                        >
                          Show 
                          3
                        </option>
                        <option
                          value={5}
                        >
                          Show 
                          5
                        </option>
                        <option
                          value={10}
                        >
                          Show 
                          10
                        </option>
                        <option
                          value={25}
                        >
                          Show 
                          25
                        </option>
                        <option
                          value={50}
                        >
                          Show 
                          50
                        </option>
                        <option
                          value={100}
                        >
                          Show 
                          100
                        </option>
                      </select>
                    </div>
                  </div>
                </td>
              </tr>
            </tfoot>
          </table>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="h-screen w-screen bg-black fixed bg-opacity-30 backdrop-blur-sm flex items-center justify-center z-50 overflow-auto mt-10 p-4 hidden"
  >
    <div
      className="bg-white dark:bg-dark-bg w-screen md:w-1/2  xl:w-4/12 rounded-lg p-4 pb-8"
    >
      <div
        className="card-title w-full flex  flex-wrap justify-center items-center  "
      >
        <h3
          className="font-bold text-xl dark:text-white text-center w-11/12"
        >
          Delete Cohort
        </h3>
        <hr
          className=" bg-primary border-b my-3 w-full"
        />
      </div>
      <div
        className="card-body"
      >
        <form
          className=" py-3 px-8"
        >
          <div>
            <h2
              className="text-base dark:text-white text-center m-4"
            >
              reallyRemoveCohort
            </h2>
          </div>
          <div
            className="w-full flex justify-between"
          >
            <button
              className="btn info sm w-[30%] md:w-1/4 text-sm font-sans"
              data-testid="delete"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
            <button
              className="btn danger sm w-[30%] md:w-1/4 text-sm font-sans"
              data-testid="confirmDeleteBtn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Delete
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="bg-light-bg dark:bg-dark-frame-bg min-h-screen"
  >
    <div
      className="flex items-left px-7 lg:px-60 pt-24 pb-8"
    >
      <div
        className="space-x-8 lg:ml-10"
      >
        <button
          className="btn primary lg undefined"
          data-testid="removeModel"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
           
          Cohort
           +
        </button>
      </div>
    </div>
    <div
      className="px-3 m d:px-8 w-screen overflow-x-auto"
    />
  </div>,
]
`;
